### PR TITLE
Add Debug info

### PR DESCRIPTION
**Pull request by dekrain**
...Thanks to (maybe) large files for hour delay

### CODE REVIEW

File: src/SLW.js [hunk 1]
```diff
@@ -188,6 +188,11 @@ export default class SLW {
       
       ctx.drawImage(Text.write(str), 4, 4)
     }
+
+    if (this.level.debug) {
+      ctx.drawImage(Text.write('x: '+this.player.x), 2, 2)
+      ctx.drawImage(Text.write('y: '+this.player.y), 2, 4)
```
Comment (bates64) [inline src/SLW.js]: These are gonna appear right next to eachother BTW.
The positions are absolute.

_Also it looks like you aren't testing your code locally before pushing?_

**Review by bates64 — CHANGES_REQUESTED** [review]
- Move the text, it'll overlap.
- Make sure you test your code.

**Review by bates64 — APPROVED** [review]

**Comment by dekrain** [comment]
I cannot test this because of error:

```
Tile.js:53:Uncaught RangeError: Tile W not found.
```

a row's "length" property is 52, but real size is 51

**Comment by dekrain** [comment]
Hmm... I tried to change CRLF to LF, but without effects...

**Comment by bates64** [comment]
> Tile.js:53:Uncaught RangeError: Tile W not found.

That means that there's a Tile in your level `W` that isn't defined. Check **src/levels.toml**.

**Comment by dekrain** [comment]
It's probably because `index.js` is encoded in CRLF format, but it should be LF.
Originally in logs are line-end character. I removed it. It's probably CR character. In my [Path2D parser](/dekrain/path-parser) I used

``` JavaScript
code.split(/[\n|\r\n]/)
```

**Comment by bates64** [comment]
That'll make no difference, we're using a build system that removes most (if not all) line breaks / carriage returns anyway.

> That means that there's a Tile in your level W that isn't defined. Check src/levels.toml.

There's a `W` in your tilemap somewhere that's confusing it at runtime **not compile-time**.

**Comment by dekrain** [comment]
Yes, it's because gulp changed `~~~~\n~~~~` to `~~~~\r\n~~~~`

**Comment by towerofnix** [comment]
I'm not honestly sure where a string of `~` would appear anyways.

The original problem is "I can't run the game". Just to clarify - can you build and run the game now?

**Comment by dekrain** [comment]
For example, in my src/levels.toml#L10

```
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~[CR][LF]
```

gulp changes it to

```
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\r\n
```

but expected is

```
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\n
```

**Comment by towerofnix** [comment]
(this message has for convenience been deleted but I see what you mean sorry lol)

**Comment by bates64** [comment]
Interesting. Out of interest, do you use Windows, @dekrain? I believe the rest of us use Linux/Mac so that's probably why you're getting a linefeed result.

**Comment by bates64** [comment]
51b79807750fa0f009b9088769045c7e9f50fdc2 should fix your bug @dekrain - merge from master and build again to see if it fixes it.

**Comment by dekrain** [comment]
Yes, I'm using Windows. Thanks for that commit.
Hmm... It seems that new plugin doesn't removes `/\\r/` sequence

**Pull request closed without merging**